testify.nim

```diff
@@ -1,74 +1,87 @@
 import os, osproc, streams, xmltree, strformat, strutils, times
 
 const
-  R = "\e[31;1m"
-  G = "\e[32;1m"
-  Y = "\e[33;1m"
-  B = "\e[34;1m"
-  D = "\e[0m"
+  stError = "\e[31;1m"
+  stSuccess = "\e[32;1m"
+  stFailure = "\e[33;1m"
+  stSuite = "\e[34;1m"
+  resetCode = "\e[0m"
 
-if paramCount() < 1:
-  echo "Too few arguments"
-  quit QuitFailure
+proc main =
+  let params = commandLineParams()
+  if "--help" in params or "-h" in params or params.len < 2:
+    quit("Compile-Run-Report helper for Nim\n\n" &
+        "Command line syntax: \n\n" &
+        "  > ./testify output_path suite_dir1, suite_dir2, ...\n")
 
-let report = newFileStream(paramStr(1), fmWrite)
-var testsuites = newElement("testsuites")
+  let
+    suites = newElement("testsuites")
+    curDir = getCurrentDir()
+    binDir = "bin"
+    nim = getCurrentCompilerExe()
 
-for d in commandLineParams()[1..^1]:
-  echo (getCurrentDir() / &"{d}")
-  setCurrentDir(getCurrentDir() / &"{d}")
-  createDir("bin")
-
-  var
-    suitename = lastPathPart(d)
-    testsuite = newElement("testsuite")
-    tests: int
-    failures: int
-    errors: int
-
-  stdout.write &"{B}[Suite]{D} " & suitename & "\n"
-
-  for f in walkFiles("t*.nim"):
-    inc(tests)
+  for i in 1 ..< params.len:
+    let suiteDir = curDir / params[i]
+    stdout.write suiteDir, "\n"
+    setCurrentDir(suiteDir)
+    createDir(binDir)
 
+    let
+      suiteName = lastPathPart(params[i])
+      suite = newElement("testsuite")
     var
-      casename = splitFile(f).name
-      testcase = newElement("testcase")
+      tests = 0
+      failures = 0
+      errors = 0
 
-    let (co, cc) = execCmdEx(&"nim c --hints:off -w:off --outdir:bin {f}")
-    if cc != 0:
-      inc(errors)
-      stdout.write &"  {R}[ER]{D} " & casename[1..^1] & "\n"
-      testcase.attrs = {"name": casename, "time": "0.00000000"}.toXmlAttributes
-      testcase.add(newXmlTree("failure", [],
```

Comment:
Should this be "error" instead?

Reply:
If I recall correctly, I needed this to be `"failure"` for my use-case. Let's not change this for now.

Reply:
That's obviously a bug but if you depend on it let's keep it.